Projects are not rebuilt if their build failed in a previous run

In this repo, when building the examples, they will fail, as there is a syntax error in one file:
```shell
rm .yarn/local-build-cache.json # in case it exists
╰─ yarn build && echo $?          
➤ YN0000: ┌ Building All
➤ YN0009: │ ┌ Errors for packages/examples/words/sit
➤ YN0009: │ │ src/index.ts:1:14 - error TS1005: ';' expected.
➤ YN0009: │ │ 1 ftygukjhu +c asda
➤ YN0009: │ │                ~~~~
➤ YN0009: │ │ Found 1 error.
➤ YN0009: │ └ End packages/examples/words/sit
➤ YN0000: └ Build finished with 2 errors
➤ YN0000: [11:2/13]
➤ YN0000: Done in 11s 899ms
0
```

If you build the projects again right after, `packages/examples/words/sit` is not built again:
```shell
╰─ yarn build && echo $?
➤ YN0000: └ Build finished
➤ YN0000: [0:0/0]
➤ YN0000: Done in 0s 167ms
0
```

--- a/packages/plugins/plugin-build/src/commands/build/supervisor.ts
+++ b/packages/plugins/plugin-build/src/commands/build/supervisor.ts
@@ -221,6 +221,10 @@ class BuildSupervisor {
     };
 
     for (const [id, entry] of this.buildLog) {
+      if (entry.status !== BuildStatus.succeeded) {
+        continue;
+      }
+
       buildLogFile.packages[id] = {
         lastModified: entry.lastModified,
       };

